scipy/stats/stats.py

```diff
@@ -802,20 +883,20 @@ def tsem(a, limits=None, inclusive=(True, True), axis=0, ddof=1):
     1.1547005383792515
 
     """
-    a = np.asarray(a).ravel()
+    a, weights, axis = _chk_weights([a], weights=weights, axis=axis)
     if limits is None:
-        return a.std(ddof=ddof) / np.sqrt(a.size)
-
+        limits = (None, None)
     am = _mask_to_limits(a, limits, inclusive)
-    sd = np.sqrt(np.ma.var(am, ddof=ddof, axis=axis))
-    return sd / np.sqrt(am.count())
+    sd = np.sqrt(_var(am, axis=axis, ddof=ddof, weights=weights))
+    n = am.count(axis=axis)
+    return sd / np.sqrt(n)
```

Comment:
Hmm that's true, I missed that one. I'm not convinced that `sd / sqrt(weights.sum())` is a valid estimate of standard error for all definitions of weights--it probably only works for `freq_weights`, so I'm inclined to remove weighting from `tsem`. Agreed?